### PR TITLE
[BULK] Support custom transformations

### PR DESCRIPTION
**Pull request by Deep1998**

### CODE REVIEW

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java [hunk 1]
```diff
@@ -107,9 +134,99 @@ public Map<String, Value> transformChangeEvent(GenericRecord record, String srcT
...
+    org.joda.time.Instant startTimestamp = org.joda.time.Instant.now();
+    MigrationTransformationRequest migrationTransformationRequest =
+        new MigrationTransformationRequest(tableName, sourceRecord, shardId, "INSERT");
+    LOG.debug(
+        "using migration transformation request {} for table {}",
+        migrationTransformationRequest,
+        tableName);
+    MigrationTransformationResponse migrationTransformationResponse =
+        customTransformer.toSpannerRow(migrationTransformationRequest);
+    org.joda.time.Instant endTimestamp = org.joda.time.Instant.now();
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java:219]: The metrics should be in a try - finally block to ensure they are always captured even there is an exception

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java [hunk 2]
```diff
@@ -128,27 +245,74 @@ public Value getSpannerValue(
...
       if (types.size() == 2 && types.stream().anyMatch(s -> s.getType().equals(Schema.Type.NULL))) {
-        fieldSchema =
-            types.stream().filter(s -> !s.getType().equals(Schema.Type.NULL)).findFirst().get();
+        return types.stream().filter(s -> !s.getType().equals(Schema.Type.NULL)).findFirst().get();
       } else {
-        throw new IllegalArgumentException(
-            String.format(
-                "Unknown schema field type %s for field %s with value %s.",
-                fieldSchema, recordColName, recordValue));
+        throw new IllegalArgumentException("Unsupported schema type");
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java:282]: can you add back the details of the error here.

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java [hunk 3]
```diff
@@ -128,27 +245,74 @@ public Value getSpannerValue(
...
+   * on whether it is a logical type or a nested record. The resulting value is potentially
+   * transformed and returned.
+   *
+   * @param recordValue The original value of the field in the GenericRecord.
+   * @param fieldSchema The Avro schema describing the field's type.
+   * @param recordColName The name of the column (field) in the record.
+   * @return The processed value of the field, potentially transformed according to its complex
+   *     type.
+   */
+  private Object handleComplexAvroTypes(
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java:301]: nit: Please use a more descriptive method name

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]

File: cicd/internal/flags/common-flags.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ const (
 // Avoid making these vars public.
 var (
 	modulesToBuild string
-	moduleMap      = map[string]string{ALL: "", SPANNER: "v2/datastream-to-spanner/,v2/spanner-change-streams-to-sharded-file-sink/,v2/gcs-to-sourcedb/,v2/sourcedb-to-spanner/,v2/spanner-custom-shard,plugins/templates-maven-plugin"}
+	moduleMap      = map[string]string{ALL: "", SPANNER: "v2/sourcedb-to-spanner/,v2/spanner-custom-shard,plugins/templates-maven-plugin"}
```
[the remark this answers is not in the document]
Comment (Deep1998) [reply cicd/internal/flags/common-flags.go:32]: This was done for quickly running ITs and debugging. Will remove

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**